app/src/main/kotlin/com/wire/android/util/CurrentScreenManager.kt

```diff
@@ -123,17 +123,22 @@ class CurrentScreenManager @Inject constructor(
     }
 
     override fun onDestinationChanged(controller: NavController, destination: NavDestination, arguments: Bundle?) {
-        val currentView = currentScreenState.value.toString()
-        AnonymousAnalyticsManagerImpl.stopView(currentView)
+        val currentScreenName = currentScreenName()
+        AnonymousAnalyticsManagerImpl.stopView(currentScreenName)
+
         val currentItem = destination.toDestination()
         currentScreenState.value = CurrentScreen.fromDestination(
             currentItem,
             arguments,
             isApplicationVisibleFlow.value
         )
 
-        val newView = currentScreenState.value.toString()
-        AnonymousAnalyticsManagerImpl.recordView(newView)
+        val newScreenName = currentScreenName()
+        AnonymousAnalyticsManagerImpl.recordView(newScreenName)
+    }
+
+    private fun currentScreenName() = currentScreenState.value.let { currentScreen ->
+        (currentScreen as? CurrentScreen.SomeOther)?.route ?: currentScreen.javaClass.simpleName
```

Comment:
Question: I might be wrong, but this will use reflection, and we should try to avoid it in Android or at least we might fallback to another option. Maybe is not a concern, but this will run constantly when Analytics enabled.